src/target/ext/thread_extra_info.rs

```diff
@@ -0,0 +1,22 @@
+//! Provide extra information for a thread
+use crate::common::Tid;
+use crate::target::{Target, TargetResult};
+
+/// Target Extension - Provide extra information for a thread
+pub trait ThreadExtraInfo: Target {
+    /// Provide extra information about a thread
+    ///
+    /// GDB queries for extra information for a thread as part of the
+    /// `info threads` command.  This function will be called once
+    /// for each active thread.
+    ///
+    /// A string can be copied into `buf` that will then be displayed
+    /// to the client.  The string is displayed as `(value)`, such as:
+    ///
+    /// `Thread 1.1 (value)`
+    ///
+    /// Return the number of bytes written into `buf`.
+    fn extra_info(&self, tid: Tid, buf: &mut [u8]) -> TargetResult<usize, Self>;
```

Comment:
reading through the docs, it seems that `qThreadExtraInfo` is _not_ allowed to return a arbitrary error value. As such, this signature should be modified to return a regular `Result<usize, Self::Error>`.

This is a common inconsistency found in the GDB RSP, where only some methods support reporting error codes, but not others. It's pretty annoying...

Reply:
Yep, you're right.  It's clear that the GDB protocol has evolved very organically so there's a lot of pitfalls.

Updated.

Reply:
I missed part of this in the second push, should all be fixed to use `Result<usize, Self::Error>` now.